addons/garage/functions/fnc_populateLists.sqf

```diff
@@ -20,25 +20,42 @@ private _display = findDisplay IDD_DISPLAY;
 // Get animation and texture data for current vehicle
 private _vehicleData = [GVAR(center)] call FUNC(getVehicleData);
 _vehicleData params ["_vehicleAnimations", "_vehicleTextures"];
+private _vehicleType = typeOf GVAR(center);
 
 private _fnc_addToList = {
-    params ["_ctrlList", "_configName", "_displayName", "_isChecked"];
+    params ["_ctrlList", "_variant", "_displayName", "_isChecked"];
 
     if (_isChecked isEqualType false) then {_isChecked = parseNumber _isChecked};
 
     private _index = _ctrlList lbAdd _displayName;
-    _ctrlList lbSetData [_index, _configName];
+    private _dataVar = str _index;
+
+    _ctrlList lbSetData [_index, _dataVar];
     _ctrlList lbSetValue [_index, _isChecked];
     _ctrlList lbSetTooltip [_index, _displayName];
     _ctrlList lbSetPicture [_index, [ICON_UNCHECKED, ICON_CHECKED] select _isChecked];
+
+    _ctrlList setVariable [_dataVar, _variant];
 };
 
 // Add items to animations list
 private _ctrlListAnimations = _display displayCtrl IDC_LIST_ANIMATIONS;
 {
     _x params ["_configName", "_displayName"];
 
-    private _isChecked = GVAR(center) animationPhase _configName;
+    private _source = getText (configFile >> "CfgVehicles" >> _vehicleType >> "animationSources" >> _configName >> "source");
+    private _phase = switch (_source) do {
+        case "door": {
+            _vehicle doorPhase _configName;
+        };
+        case "user": {
+            _vehicle animationSourcePhase _configName;
+        };
+        default {
+            _vehicle animationPhase _configName;
+        };
```

Comment:
`_vehicle` is not defined in this function. Works currently because its called `openGarage` function.